sdks/python/apache_beam/runners/interactive/interactive_beam_test.py

```diff
@@ -0,0 +1,70 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Tests for apache_beam.runners.interactive.interactive_beam."""
+
+import importlib
+import unittest
+
+from apache_beam.runners.interactive import interactive_beam as ib
+from apache_beam.runners.interactive import interactive_environment as ie
+
+# The module name is also a variable in module.
+_module_name = 'apache_beam.runners.interactive.interactive_beam_test'
+
+
+class InteractiveBeamTest(unittest.TestCase):
+
+  def setUp(self):
+    self._var_in_class_instance = 'a var in class instance, not directly used'
+    ie.new_env()
+
+  def test_watch_main_by_default(self):
+    test_env = ie.InteractiveEnvironment()
+    # Current Interactive Beam env fetched and the test env are 2 instances.
+    self.assertNotEqual(id(ie.current_env()), id(test_env))
+    self.assertEqual(ie.current_env().watching(), test_env.watching())
+
+  def test_watch_a_module_by_name(self):
+    test_env = ie.InteractiveEnvironment()
+    ib.watch(_module_name)
+    test_env.watch(_module_name)
+    self.assertEqual(ie.current_env().watching(), test_env.watching())
```

Comment:
What is really tested here? We call watch on two environments and check that they have the same results. It is not checking whether the _module_name is in the watch set or not as the test name implies. (Same comment for other assertEqual's in this file.)

Reply:
These tests verifies that the static watch() function in interactive_beam module will eventually generate the same watching() items to the member watch() function in interactive_environment module InteractiveEnvironment class when the correct InteractiveEnvironment object is accessed.
Whatever future changes happen to the wrapper in interactiv_beam module, it should keep this behavior.
It currently basically tests if "ie.current_env().watch(watchable)" works as intended: create environment if absent, access current environment and invoke the watch() logic for that environment. 
It doesn't verify if "watch(watchable)" generates expected "watching()" because that is not the logic of this module.

The unit tests test the wrapped watch() logic in interactive_beam module where the InteractiveEnvironment objects may be created and accessed. The real watch() logic within an InteractiveEnvironment object from interactive_environment module is tested in its own unit test.

Reply:
Sounds good.